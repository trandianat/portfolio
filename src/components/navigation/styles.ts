import { css } from '@emotion/react';
import { Color } from 'utils/constants';

export const navigation = css({
  backgroundColor: Color.BLACK,
  border: `1px solid ${Color.BLACK}`,
  boxShadow: `0 4px 8px -2px ${Color.DARK_GRAY}`,
  display: 'flex',
  margin: '-48px 0 36px',
  position: 'sticky',
  textAlign: 'center',
  top: 0,
  a: {
    color: Color.WHITE,
    flexGrow: 1,
    fontWeight: 'normal',
    padding: '12px 0',
    textDecoration: 'none',
    svg: { fill: Color.WHITE },
    '&.active': {
      backgroundColor: Color.WHITE,
      color: Color.BLACK,
      fontWeight: 500,
      pointerEvents: 'none',
    },
    ':hover': {
      backgroundColor: Color.WHITE,
      color: Color.BLACK,
      svg: { fill: Color.BLACK },
    },
  },
  '@media (max-width: 700px)': {
    fontSize: '.875rem',
    a: { ':last-child': { display: 'none' } },
  },
  '@media (max-width: 450px)': {
    flexDirection: 'column',
    a: {
      letterSpacing: 4,
      padding: '10px 0',
      ':nth-of-type(-n + 2)': { borderBottom: `1px solid ${Color.WHITE}` },
      '&.active': { fontWeight: 600 },
    },
  },
});

export const external = {
  alignItems: 'center',
  display: 'flex',
  gap: 6,
  justifyContent: 'center',
};
