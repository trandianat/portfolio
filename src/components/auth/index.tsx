import { useAuthenticator, withAuthenticator } from '@aws-amplify/ui-react';
import { Fragment, ReactNode } from 'react';
import { Link, useLocation } from 'react-router-dom';
import * as styles from 'components/auth/styles';
import isProjects from 'utils/projects';

const Auth = ({ children }: { children: ReactNode }): JSX.Element => {
  const { signOut } = useAuthenticator();
  const { pathname } = useLocation();
  return (
    <Fragment>
      <div css={styles.navigation}>
        {isProjects(pathname) ? <div /> : <Link to="projects">â‰ª Back</Link>}
        <button onClick={signOut}>Sign out</button>
      </div>
      {children}
    </Fragment>
  );
};

export default withAuthenticator(Auth, {
  components: { SignIn: { Footer: () => <Fragment /> } },
  formFields: {
    signIn: {
      username: { label: '', placeholder: 'Username' },
      password: { label: '', placeholder: 'Password' },
    },
  },
  hideSignUp: true,
  // services: {
  //   handleSignIn: async ({
  //     username,
  //     password,
  //   }: {
  //     username: string;
  //     password: string;
  //   }) => console.log('Credentials: ' + username + ' / ' + password),
  // },
});
