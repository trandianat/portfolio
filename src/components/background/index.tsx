import { PropsWithChildren } from 'react';
import * as styles from 'components/background/styles';
import { Color } from 'utils/types';

export const Background = ({
  children,
  color = Color.WHITE,
}: PropsWithChildren<{ color?: Color }>): JSX.Element => {
  return (
    <div css={styles.background(color)}>
      <div css={styles.container}>{children}</div>
    </div>
  );
};
