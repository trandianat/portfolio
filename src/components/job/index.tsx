import * as styles from 'components/job/styles';

export const Job = ({
  role,
  company,
  start,
  end,
  skills,
  descriptions,
  indent = false,
}: {
  role: string;
  company: string;
  start: string;
  end: string;
  skills: string[];
  descriptions: string[];
  indent?: boolean;
}): JSX.Element => (
  <div css={styles.job(indent)}>
    <div css={styles.dates}>
      <div css={styles.range}>
        <div className="sans">{end}</div>
        <div className="sans">{start}</div>
      </div>
      <div css={styles.timeline}>
        <div css={styles.circle} />
        <div css={styles.line} />
        <div css={styles.circle} />
      </div>
    </div>
    <div css={styles.role}>
      <h2>{role}</h2>
      <p className="sans">{company}</p>
      <div css={styles.skills}>
        {skills.map((skill: string) => (
          <div css={styles.skill} key={skill}>
            {skill}
          </div>
        ))}
      </div>
      <ul>
        {descriptions.map((description: string) => (
          <li key={description}>{description}</li>
        ))}
      </ul>
    </div>
  </div>
);
