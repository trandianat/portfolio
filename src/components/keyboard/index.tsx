import { useEffect, useState } from 'react';
import * as styles from 'components/keyboard/styles';
import { Keyboard as Key } from 'utils/constants';

export const Keyboard = ({
  disabled,
  guesses,
  handleSubmit,
  input,
  setInput,
}: {
  disabled: boolean;
  guesses: string[];
  handleSubmit: any;
  input: string;
  setInput: any;
}): JSX.Element => {
  const [rows, setRows] = useState<Array<string[]>>([]);

  const isAction = (key: string): boolean =>
    [Key.DELETE, Key.ENTER].includes(key as Key);
  const isDelete = (key: string): boolean => key === Key.DELETE;
  const isEnter = (key: string): boolean => key === Key.ENTER;

  const setLetterClass = (key: string, rowIndex: number): string => {
    let className = 'letter';
    if (isAction(key)) {
      className += ' action';
      if (rowIndex < rows.length - 1) {
        className += ' primary';
      }
    } else {
      const used = guesses.map(guess => [...guess]).flat();
      if (used.includes(key)) {
        className += ' used';
      }
    }
    return className;
  };

  useEffect(() => {
    const row1 = ['Q', 'W', 'E', 'R', 'T', 'Y', 'U', 'I', 'O', 'P'];
    const row2 = ['A', 'S', 'D', 'F', 'G', 'H', 'J', 'K', 'L'];
    const row3 = [Key.ENTER, 'Z', 'X', 'C', 'V', 'B', 'N', 'M', Key.DELETE];
    const row4 = [Key.ENTER, Key.DELETE];
    setRows([row1, row2, row3, row4]);
  }, []);

  return (
    <div css={styles.keyboard}>
      {rows.map((row, index) => (
        <div className={`row row-${index + 1}`} key={`row-${index + 1}`}>
          {row.map((key: string) => (
            <button
              className={setLetterClass(key, index)}
              key={key}
              onClick={() => {
                if (isDelete(key)) {
                  setInput((input: string) => input.slice(0, -1));
                } else if (isEnter(key)) {
                  handleSubmit();
                } else {
                  setInput((input: string) => input + key);
                }
              }}
              {...(((isDelete(key) && !input.length) ||
                (isEnter(key) && input.length < 5) ||
                (!isAction(key) && input.length === 5) ||
                disabled) && {
                disabled: true,
              })}
            >
              {key}
            </button>
          ))}
        </div>
      ))}
    </div>
  );
};
