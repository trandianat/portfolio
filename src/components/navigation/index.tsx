import { Link, useLocation } from 'react-router-dom';
import LinkIcon from 'assets/icons/link';
import { Link as LinkOut } from 'components/link';
import * as styles from 'components/navigation/styles';
import { Links, Position, Size } from 'utils/constants';

export const Navigation = (): JSX.Element => {
  const { pathname } = useLocation();
  const pages = [
    { name: 'Experience', path: '/' },
    { name: 'Skills', path: '/skills' },
    { name: 'Projects', path: '/projects' },
  ];

  const isExperience = (path: string): boolean =>
    path === '/' && !pathname.substring(1);

  const isParentPage = (path: string): boolean =>
    path !== '/' && path === pathname;

  const isSubPage = (path: string): boolean => {
    const pathWithoutSlash = path.substring(1);
    const page = pathname.substring(1);
    return (
      !!pathWithoutSlash &&
      pathWithoutSlash === page.substring(0, page.indexOf('/'))
    );
  };

  return (
    <nav className="sans" css={styles.navigation}>
      {pages.map(({ name, path }: { name: string; path: string }) => (
        <Link
          key={name}
          to={path}
          {...((isExperience(path) ||
            isParentPage(path) ||
            isSubPage(path)) && {
            className: 'active',
          })}
        >
          {name}
        </Link>
      ))}
      <LinkOut
        css={styles.external}
        icon={<LinkIcon />}
        iconGap={Size.LARGE}
        iconPosition={Position.LEFT}
        text="Github"
        url={Links.GITHUB}
      />
    </nav>
  );
};
