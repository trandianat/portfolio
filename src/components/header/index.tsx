import LinkOut from 'assets/icons/link';
import { Background } from 'components/background';
import * as styles from 'components/header/styles';
import { Link } from 'components/link';
import { Links, Position, Size } from 'utils/constants';

export const Header = (): JSX.Element => (
  <Background>
    <div css={styles.header}>
      <div css={styles.divider} />
      <div css={styles.name}>
        <div>
          <h1>Diana Tran</h1>
          <p className="sans">Software Engineer</p>
        </div>
        <div css={styles.logo}>
          <div className="content">
            <div className="dot" />
            <div>DT</div>
            <div className="line" />
          </div>
        </div>
      </div>
      <p>
        Results-driven team player with 8 years of full-stack experience and 1
        year of management experience. Detail-oriented documentation expert who
        is eager to fail fast, learn, and iterate. Well versed in the software
        development lifecycle including design thinking, architecture and system
        design, testing, and monitoring. Graduated from Boston College in 2014
        with a bachelor's degree in Computer Science.{' '}
        <span css={styles.links}>
          <Link
            icon={<LinkOut />}
            iconGap={Size.MEDIUM}
            iconPosition={Position.RIGHT}
            text="LinkedIn"
            url={Links.LINKEDIN}
          />{' '}
          &{' '}
          <Link
            icon={<LinkOut />}
            iconGap={Size.MEDIUM}
            iconPosition={Position.RIGHT}
            text="Github"
            url={Links.GITHUB}
          />
        </span>
      </p>
    </div>
  </Background>
);
