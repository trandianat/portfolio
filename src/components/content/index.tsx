import { Fragment, ReactNode } from 'react';
import { Background } from 'components/background';
import { Header } from 'components/header';
import { Navigation } from 'components/navigation';
import { Color } from 'utils/constants';

export const Content = ({ children }: { children: ReactNode }): JSX.Element => (
  <Fragment>
    <Header />
    <Background color={Color.GRAY}>
      <Navigation />
      {children}
    </Background>
  </Fragment>
);
