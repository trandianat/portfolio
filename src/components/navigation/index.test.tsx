import { ReactNode } from 'react';
import { useLocation } from 'react-router-dom';
import { create } from 'react-test-renderer';
import { Navigation } from 'components/navigation';

jest.mock('react-router-dom', () => ({
  Link: ({ children, to }: { children: ReactNode; to: string }) => (
    <a href={to}>{children}</a>
  ),
  useLocation: jest.fn(),
}));

describe('Navigation', () => {
  const mockUseLocation = useLocation as jest.Mock;
  mockUseLocation.mockReturnValue({ pathname: '/' });

  it('should match the snapshot', () => {
    expect(create(<Navigation />)).toMatchSnapshot();
  });
});
