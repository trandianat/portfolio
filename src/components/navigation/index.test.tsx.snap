// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation should match the snapshot 1`] = `
.emotion-0 {
  background-color: var(--black);
  border: 1px solid var(--black);
  box-shadow: var(--shadow);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: -48px 0 36px;
  position: -webkit-sticky;
  position: sticky;
  text-align: center;
  top: 0;
  z-index: 1;
}

.emotion-0 a {
  color: var(--white);
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  font-weight: normal;
  padding: 12px 0;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-0 a svg {
  fill: var(--white);
}

.emotion-0 a.active {
  background-color: var(--white);
  color: var(--black);
  font-weight: 500;
  pointer-events: none;
}

.emotion-0 a:hover {
  background-color: var(--white);
  color: var(--black);
}

.emotion-0 a:hover svg {
  fill: var(--black);
}

@media (max-width: 950px) {
  .emotion-0 .external {
    display: none;
  }
}

@media (max-width: 700px) {
  .emotion-0 a {
    font-size: .875rem;
  }
}

@media (max-width: 450px) {
  .emotion-0 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }

  .emotion-0 a {
    letter-spacing: 4px;
    padding: 10px 0;
  }

  .emotion-0 a:nth-of-type(-n + 2) {
    border-bottom: 1px solid var(--white);
  }
}

.emotion-1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  gap: 6px;
  vertical-align: bottom;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

<nav
  className="sans emotion-0"
>
  <a
    href="/"
  >
    Experience
  </a>
  <a
    href="/skills"
  >
    Skills
  </a>
  <a
    href="/projects"
  >
    Projects
  </a>
  <a
    className="external emotion-1"
    href="https://www.linkedin.com/in/trandianat"
    target="_blank"
  >
    <svg
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M5 3c-1.093 0-2 .907-2 2v14c0 1.093.907 2 2 2h14c1.093 0 2-.907 2-2v-7h-2v7H5V5h7V3H5zm9 0v2h3.586l-9.293 9.293 1.414 1.414L19 6.414V10h2V3h-7z"
      />
    </svg>
    LinkedIn
  </a>
  <a
    className="external emotion-1"
    href="https://github.com/trandianat"
    target="_blank"
  >
    <svg
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M5 3c-1.093 0-2 .907-2 2v14c0 1.093.907 2 2 2h14c1.093 0 2-.907 2-2v-7h-2v7H5V5h7V3H5zm9 0v2h3.586l-9.293 9.293 1.414 1.414L19 6.414V10h2V3h-7z"
      />
    </svg>
    Github
  </a>
</nav>
`;
